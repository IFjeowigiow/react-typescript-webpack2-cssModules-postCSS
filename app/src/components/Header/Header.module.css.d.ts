// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'header': string;
  'headerBg': string;
  'header_bg': string;
}
export const cssExports: CssExports;
export default cssExports;
