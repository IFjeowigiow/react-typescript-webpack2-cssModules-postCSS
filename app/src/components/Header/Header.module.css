@import "../../../stylesheets/1_Settings/_settings.colors.css";

/*
  Exports the variable to be exposed in CSS Modules (styles object in React)
*/
:export {
  header_bg: var(--headerBackground);
}

.header {
  text-align: center;
  font-size: .8em;
  padding-bottom: 1px;
  margin-bottom: 5px;
  background-color: var(--headerBackground);
}


