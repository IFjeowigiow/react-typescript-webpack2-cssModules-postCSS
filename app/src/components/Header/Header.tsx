import * as React from "react";

/* tslint:disable-next-line:no-var-requires */
const styles: any = require("./Header.module.css");

import { IHeaderProps } from "./IHeaderProps";

class Header extends React.Component<IHeaderProps, {}> {

    public render() {
        return (
            <header className={styles.header}>
                <h1>Hello, {this.props.title}</h1>
                <h2>Background Color from PostCSS variable: {styles.header_bg}</h2>
            </header>
        );
    }
}

export default Header;
