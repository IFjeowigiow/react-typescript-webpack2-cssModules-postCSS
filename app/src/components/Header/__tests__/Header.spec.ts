import testComponentHelper from '../../../../../test/unit/helpers/ComponentHelper'
import Header from './../Header'

describe('<Header />', () => {
    const initialProps = {
        className: '',
    };
    const renderComponent = testComponentHelper(Header);

    describe('@renders', () => {
        it('in default state', () => {
            expect(renderComponent().getHtml())
                .toMatchSnapshot();
        })
    })

    // describe('Header presentational component', () =>{
    //     it('should renders two Link elements, first one with propert to equals "/about" and ' +
    //         'text equals "About" and second one with property to equals "/members" and text' +
    //         'equals "Members"', () => {
    //         let headerWrapper = shallow(<Header />);
    
    //         expect(headerWrapper.find('Link').at(0)).to.be.exist;
    //         expect(headerWrapper.find('Link').at(0).prop('to')).to.be.equals("/about");
    //         expect(headerWrapper.find('Link').at(0).children().text()).to.be.equals("About");
    //         expect(headerWrapper.find('Link').at(1)).to.be.exist;
    //         expect(headerWrapper.find('Link').at(1).prop('to')).to.be.equals("/members");
    //         expect(headerWrapper.find('Link').at(1).children().text()).to.be.equals("Members");
    //     });
    // })


})