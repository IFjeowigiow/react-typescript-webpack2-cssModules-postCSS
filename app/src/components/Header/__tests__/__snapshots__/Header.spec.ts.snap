// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> @renders in default state 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <header
    className="header"
  >
    <h1>
      Hello, 
    </h1>
    <h3>
      Header background Color from PostCSS =&gt; var(--headerBackground): 
      header_bg
    </h3>
  </header>,
  "nodes": Array [
    <header
      className="header"
    >
      <h1>
        Hello, 
      </h1>
      <h3>
        Header background Color from PostCSS =&gt; var(--headerBackground): 
        header_bg
      </h3>
    </header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header>
         
         
      </Header>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": Array [
            " ",
            null,
            " ",
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <header
          className="header"
        >
          <h1>
            Hello, 
          </h1>
          <h3>
            Header background Color from PostCSS =&gt; var(--headerBackground): 
            header_bg
          </h3>
        </header>,
        "_debugID": 2,
        "_renderedOutput": <header
          className="header"
        >
          <h1>
            Hello, 
          </h1>
          <h3>
            Header background Color from PostCSS =&gt; var(--headerBackground): 
            header_bg
          </h3>
        </header>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header>
     
     
  </Header>,
}
`;
